domain/schema/controller.go

```diff
@@ -445,3 +448,29 @@ CREATE TABLE model_defaults (
 );
 `)
 }
+
+func autocertCacheSchema() schema.Patch {
+	return schema.MakePatch(`
+CREATE TABLE autocert_cache (
+    uuid 		TEXT PRIMARY KEY,
+    data 		TEXT NOT NULL,
+    encoding_uuid       TEXT NOT NULL,
+    CONSTRAINT 		fk_controller_cache_encoding
+        FOREIGN KEY 	    (encoding_uuid)
+	REFERENCES 	    controller_cache_encoding(uuid)
+);
+
+-- NOTE(nvinuesa): This table only populated with *one* hard-coded value
+-- (x509) because golang's autocert cache doesn't provide encoding in it's
+-- function signatures, and in juju we are only using x509 certs. The value
+-- of this table is to correctly represent the domain and already have a
+-- list of possible encodings when we update our code in the future.
+CREATE TABLE autocert_cache_encoding (
+    uuid          TEXT PRIMARY KEY,
+    name          TEXT NOT NULL
+);
+
+INSERT INTO autocert_cache_encoding VALUES
+    (0, 'x509');    -- Only x509 certs encoding supported today.
```

Comment:
`0` is definitely not a 'uuid' text, but maybe you want to make this an `id INTEGER PRIMARY KEY` [AUTOINCREMENT?]

And then you would also need to update the other table for encoding_id